src/helpers/spaces.ts

```diff
@@ -2,32 +2,68 @@ import snapshot from '@snapshot-labs/snapshot.js';
 import { uniq } from 'lodash';
 import db from './mysql';
 import log from './log';
+import { flaggedSpaces, verifiedSpaces } from './moderation';
 
 export let spaces = {};
 export const spacesMetadata = {};
-export const spaceProposals = {};
-export const spaceVotes = {};
-export const spaceFollowers = {};
+export let rankedSpaces: any = [];
+const spaceProposals = {};
+const spaceVotes = {};
+const spaceFollowers = {};
+
+function getPopularity(id: string, verified: boolean): number {
+  let popularity =
+    (spaceVotes[id]?.count || 0) / 50 +
+    (spaceVotes[id]?.count_7d || 0) +
+    (spaceProposals[id]?.count_7d || 0) * 50 +
+    (spaceFollowers[id]?.count_7d || 0);
+
+  if (verified) {
+    popularity *= 5;
+    popularity += 100;
+  }
+
+  return popularity;
+}
 
 function mapSpaces() {
   Object.entries(spaces).forEach(([id, space]: any) => {
+    const verified = verifiedSpaces?.includes(id) || false;
+    const flagged = flaggedSpaces?.includes(id) || false;
+    const popularity = getPopularity(id, verified);
+
     spacesMetadata[id] = {
+      id,
       name: space.name,
-      private: space.private || undefined,
-      terms: space.terms || undefined,
-      network: space.network || undefined,
-      networks: uniq((space.strategies || []).map(strategy => strategy.network || space.network)),
-      categories: space.categories || undefined,
-      activeProposals: (spaceProposals[id] && spaceProposals[id].active) || undefined,
-      proposals: (spaceProposals[id] && spaceProposals[id].count) || undefined,
-      proposals_active: (spaceProposals[id] && spaceProposals[id].active) || undefined,
-      proposals_7d: (spaceProposals[id] && spaceProposals[id].count_7d) || undefined,
-      votes: (spaceVotes[id] && spaceVotes[id].count) || undefined,
-      votes_7d: (spaceVotes[id] && spaceVotes[id].count_7d) || undefined,
-      followers: (spaceFollowers[id] && spaceFollowers[id].count) || undefined,
-      followers_7d: (spaceFollowers[id] && spaceFollowers[id].count_7d) || undefined
+      verified,
+      flagged,
+      popularity,
+      private: space.private ?? false,
+      categories: space.categories ?? [],
+      networks: uniq(
+        (space.strategies || [])
+          .map(strategy => strategy?.network || space.network)
+          .concat(space.network)
+      ),
+      counts: {
+        activeProposals: spaceProposals[id]?.active || 0,
+        proposalsCount: spaceFollowers[id]?.count || 0,
+        proposalsCount7d: spaceProposals[id]?.count_7d || 0,
+        followersCount: spaceProposals[id]?.count || 0,
+        followersCount7d: spaceFollowers[id]?.count_7d || 0,
+        votesCount: spaceVotes[id]?.count || 0,
+        votesCount7d: spaceVotes[id]?.count_7d || 0
+      }
     };
   });
+
+  rankedSpaces = Object.values(spacesMetadata)
+    .sort((a: any, b: any) => b.popularity - a.popularity)
+    .filter((space: any) => !space.private);
```

Comment:
Should we hide suspicious spaces as well?

Reply:
Yea there is no confirmation from @bonustrack Maybe we can hide them in future, for this PR we don't need to hide flagged spaces